decidim-proposals/lib/decidim/proposals/component.rb

```diff
@@ -143,9 +143,11 @@
 
   component.exports :proposal_comments do |exports|
     exports.collection do |component_instance|
-      Decidim::Comments::Export.comments_for_resource(
-        Decidim::Proposals::Proposal, component_instance
-      )
+      if proposal_hidden?
+        Decidim::Comments::Export.comments_for_resource(
+          Decidim::Proposals::Proposal, component_instance
+        )
+      end
```

Comment:
⚠️ This exports proposal comments, it's not the section you should change from this file.

Also, please make sure you don't make innecessary modifications to the code (many files in this PR have style-related changes that distract attention from the relevant functionality-related changes)